### PR TITLE
memcache: SingleFlight, Cache, and Store

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: x/memcache/singleflight.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	}
+
+	return sf
+}
+
+// revive:disable:cognitive-complexity
+
+// Get attempts to get the value of a key from its internal cache, otherwise reaches
+// out to the provided [cache.Getter], but only once. While this is in process any other
+// request for the same key will be help until we have a response from from the first.
```
Comment (karasz) [inline x/memcache/singleflight.go:59]: will be **held**, not help

File: x/memcache/singleflight.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+func (sf *SingleFlight) Get(ctx context.Context, key string, dest cache.Sink) error {
+	// revive:enable:cognitive-complexity
+	sf.mu.Lock()
+
+	if v, ex, ok := sf.inward.Get(key); ok {
+		// cache hit
+		defer sf.mu.Lock()
```
[the remark this answers is not in the document]
Comment (amery) [reply x/memcache/singleflight.go:68]: sh*t

[the remark this answers is not in the document]
Comment (amery) [reply x/memcache/singleflight.go]: amended

File: x/memcache/singleflight.go (hunk 1 above)
Comment (amery) [reply x/memcache/singleflight.go]: amended

**Comment by amery** [comment]
> see comments

great catches, amended accordingly. @karasz

**Comment by amery** [comment]
`x/memcache/v0.0.2` tagged @karasz

**Pull request merged**